### PR TITLE
Add thumbnail image on mooc

### PR DESCRIPTION
**Pull request by JeremieSamson**
Permettre l'upload d'une vignette lors de la création d'un MOOC

### CODE REVIEW

File: src/Entity/Mooc/Mooc.php [hunk 1]
```diff
@@ -125,6 +121,19 @@ class Mooc
...
 
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\Image(
+     *     mimeTypes={"image/jpeg", "image/png"},
+     *     maxSize="1M",
+     *     maxWidth="480",
+     *     maxHeight="360",
+     *     minWidth="211"
```
[the remark this answers is not in the document]
Comment (JeremieSamson) [reply src/Entity/Mooc/Mooc.php]: yep, confirmed by po

File: src/Admin/MoocAdmin.php [hunk 2]
```diff
@@ -158,15 +152,23 @@ protected function configureListFields(ListMapper $listMapper): void
         ;
     }
 
-    protected function configureRoutes(RouteCollection $collection): void
+    private function addImageSrc(): ?string
+    {
+        return $this->getSubject() && $this->getSubject()->getImageName()
+            ? '<br/>><img src="'.$this->getSubject()->getAssetImagePath().'" />'
```
Comment (thomasbisignani) [inline src/Admin/MoocAdmin.php]: There is a useless `>` char : 
```php 
<br/>><img src
``` 
=>
```php
<br/><img src
```
What about using a method (`getImageHelpMessage()`) to directly return the image help message with the img tag if an image is uploaded  ?

File: src/Normalizer/MoocListNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private $router;
+
+    public function __construct(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    public function normalize($object, $format = null, array $context = [])
+    {
+        return $this->normalizeMooc($object);
```
Comment (ottaviano) [inline src/Normalizer/MoocListNormalizer.php]: why do you add a new private method?

File: src/Normalizer/MoocListNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    public function supportsNormalization($data, $format = null)
+    {
+        return self::FORMAT === $format && $data instanceof Mooc;
+    }
+
+    private function normalizeMooc(Mooc $mooc): array
+    {
+        return [
+            'chapter_count' => $mooc->countChapters(),
```
Comment (ottaviano) [inline src/Normalizer/MoocListNormalizer.php]: I think you don't need to manage other fields, but just `image`. You can decorate ItemNormalizer of ApiPlatform to do this.

File: templates/form_theme_admin.html.twig [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{% extends '@SonataAdmin/Form/form_admin_fields.html.twig' %}
+
+{% block app_admin_mooc_mooc_image_file_image_row -%}
+    {{ block('admin_image_with_url_row') }}
+{%- endblock app_admin_mooc_mooc_image_file_image_row %}
```
Comment (ottaviano) [inline templates/form_theme_admin.html.twig]: can you remove the block name here pls, as the second endblock L14 is without the name?

File: templates/form_theme_admin.html.twig [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+{% block admin_image_with_url_row %}
+    {{ block('form_row') }}
+
+    {% set object = form.parent.vars.value %}
+    {% if object.imageName is defined and object.imageName is not null %}
```
Comment (ottaviano) [inline templates/form_theme_admin.html.twig]: `is not null` => `is not empty`

**Comment by thomasbisignani** [comment]
:ok_hand:

**Pull request merged**